crates/volta-core/src/tool/mod.rs

```diff
@@ -53,11 +52,11 @@ fn info_project_version<T: Display + Sized>(tool: T) {
 /// Trait representing all of the actions that can be taken with a tool
 pub trait Tool: Display {
     /// Fetch a Tool into the local inventory
-    fn fetch(self, session: &mut Session) -> Fallible<()>;
+    fn fetch(self: Box<Self>, session: &mut Session) -> Fallible<()>;
```

Comment:
Question for my own edification: does Clippy not suggest `Box<dyn Self>` here? It seems vaguely redundant but I’m not positive on the implications off the top of my head (or even whether that’s syntactically valid!).

Reply:
It doesn't, and a quick test shows that apparently `Box<dyn Self>` is a syntax error because `Self` is a type, not a trait (or something like that).

I actually got to `Box<Self>` by first doing `Box<dyn Tool>`, which was itself an error that suggested using `Box<Self>` 😄